frontend/catalyst/api_extensions/callbacks.py

```diff
@@ -38,6 +40,64 @@
 from catalyst.utils.types import convert_pytype_to_shaped_array
 
 
+def accelerate(func=None, dev=None):
+    """Seamless integration with jax.jit
+
+    The accelerate function can be used to automatically apply `jax.jit` to
+    the input function `func` and be used to specify the device with the `dev`
+    keyword argument. If `dev` is None, then the default device as determined
+    by JAX is used.
+
+    Can be used in the following modes:
+
+
+    ```py
+    @accelerate
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    ```
+    @accelerate(dev=jax.devices()[0])
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    or
+
+    ```
+    @qml.qjit
+    def qjitted_fn(x):
+      return accelerate(lambda x: x)
+    ```
+    """
+
+    if isinstance(func, Callable):
+
+        @functools.wraps(func)
+        def defer(*args, **kwargs):
+            # Make abstract variables from input tracers.
+            absargs, abskwargs = tree_map(shaped_abstractify, (args, kwargs))
+            # Find the shape of the return value
+            _, returnshape = jax.make_jaxpr(func, return_shape=True)(*absargs, **abskwargs)
+            jitted_fn = jax.jit(func, device=dev)
+            return pure_callback(jitted_fn, result_type=returnshape)(*args, **kwargs)
+
+        return defer
+
+    kwargs = copy.copy(locals())
+    kwargs.pop("func")
+    return functools.partial(accelerate, **kwargs)
```

Comment:
@erick-xanadu it might be better here to:

- explicitly check if the function has been jitted (e.g., it is the right jax type)
- if not, raise an error (`ValueError: func must be a callable function or a JAX JIT-compiled function`)

Reply:
Why do we need this check? I think with `Callable` is enough.

Reply:
From what I can tell from reading the source code, this will result in functools partialling any non-callable? Whereas we are assuming that this logic would apply only to JITted functions?

Reply:
It is keeping track of the device. Because we also want to keep the possibility of the following syntax:

```python
@accelerate(dev)
def foo(): ...
```